### PR TITLE
Adds the NT BR-38 Battle Rifle. A sci-fi hybrid gun, but better than the last attempt. Only SOMETIMES fails at existence. 

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

A repeat of the first half of https://github.com/tgstation/tgstation/pull/86853

The NT BR-38 Battle Rifle is a semi-automatic railgun...marksman...carbine...rifle...thingy that uses .38 ammunition and is magazine fed from a 15 round magazine. It comes with a scope, and is bulky.

image

Sprites by OrcaCora.

### Some of its features

The gun shoots at a higher than normal velocity for .38. This means it hits harder (30 damage compared to the standard 25), and flies faster through the air.

The gun, however, suffers from degradation as it is used. It has a 10 shot buffer before suffering degradation. After the 10 shot buffer, it has a 10% chance each shot to increase in degradation stage. From stage 1 to stage 5 (the maximum stage), it begins to lose fire rate and projectile flight speed.

To recalibrate the gun, you can do one of two things; 
A) Use a multitool on the gun. This completely resets the gun, but its pretty slow.
B) Insert it into a weapon recharger to recharge its buffer and reduce its degradation stages.

### Some of its downsides

The weapon is EMP vulnerable. If it gets EMP'd, it immediately degrades and loses any buffer it has. While EMP'd, there is a 75% chance it doesn't fire when you pull the trigger. This is also true if the gun has hit maximum degradation. Keep your guns in good shape.

It can be emagged. While emagged, the gun has increased damage (40 for standard ammunition), but once it hits maximum degradation, it immediately begins to catastrophically fail. It also degrades significantly faster while emagged. There is no way to reverse this effect, and the explosion is extremely lethal.

### Where can I find it? What about ammo?

You can purchase the gun from cargo at a significant premium, as well as some additional magazines of some basic ammo types.

There is exactly one of these in the armory, and you can print more magazines from the security protolathe. As research progresses, you can print different, specialized ammunition for the rifle. (and also the detective .38 revolver, obviously)

### New .38 ammunition types

True Strike bounces accurately between targets, but deals significantly less damage than other ammo types (15 base, 18 in this rifle).

It is printed once the station gets Exotic Ammo.

## Why It's Good For The Game

This is a followup on my previous PR, where I tested this weapon and gathered a whole bunch of feedback on it. Most people quite liked it. but had some concerns that I feel this addresses. 

Much of the justifications given there apply here. Obviously ignoring anything to do with the combat shotgun removal.

Some stuff obviously has not come back from the original test. Notably, the TRAC changes were probably a little too powerful. As much as I liked a default entirely nonlethal ammunition type, TRAC itself is better left as a late research goal.

In addition, I moved the explosion part to an emag effect. It's funny, and lends itself to the original intention of the weapon; a callback to the WT-550 incident. But it isn't a standard part of the guns mechanics.

## Changelog
:cl: NecromancerAnne (code), orcacora (sprites)
adds: Adds NT BR-38 Battle Rifles. A hybrid weapon. Find it in your local armory and cargo catalogue today. (Keep away from EMPs)
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Don't even think about merging this before https://github.com/tgstation/tgstation/pull/87740

**Comment by Loafin34** [comment]
30 ROUNDS!!! 2 ROUND BURST FIRE!!! PHASIC!!!!!!!!!!!!!!!!!!!!! THE C20r IS IN THE HOUSE!!!!!!!!!

**Comment by necromanceranne** [comment]
>1, this behaves like a crew-powered LMG. 

The designations expand further.

>2, phasic rounds and heartseeker are probably meant for nukies only? Haven't really seen heartseeker in practice though. the phasic can go through walls, and ignores armor. It's probably one of the best rounds in the arsenal, and it's casually on this.

It's research-gated. You're not getting these rounds from the get-go.

It also isn't the only example of phasic-like rounds either. X-ray lasers, phasic junk rounds and metal hydrogen bolts are types of phasing bullets that crew can get. Some are a easier or harder to get than others. 

There aren't any homing projectiles just yet, but it's something I'd like to see in action to see how it pans out. Could be cool.

**Comment by necromanceranne** [comment]
Security officers shooting down the door to set off a maxcap for their wallhack bullets sounds funny as fuck and something I'm not terribly worried about. :U

**Comment by mc-oofert** [comment]
an even more tacticool gun than the former that is much like its predecessor is not any better than grabbing an actual laser gun due to its degradation mechanic

**Comment by necromanceranne** [comment]
> an even more tacticool gun than the former (the sprite looks like ass but the inhands is pretty decent) that is much like its predecessor is not any better than grabbing an actual laser gun due to its degradation mechanic

Okay.

**Comment by Formerxboxgamer** [comment]
> an even more tacticool gun than the former (the sprite looks like ass but the inhands is pretty decent) that is much like its predecessor is not any better than grabbing an actual laser gun due to its degradation mechanic

This player has no idea just how good a scope is and it shows. The previous BR and probably this thing will fill a very unique niche in the security arsenal by being the hallway hold king. I’m excited to see it return because while yeah a laser is probably easier to run and gun with, the BR has an advantage with range and wounding

**Comment by carlarctg** [comment]
looks awesome thumbs upp

> Security officers shooting down the door to set off a maxcap for their wallhack bullets sounds funny as fuck and something I'm not terribly worried about. :U

real gamers know u dont even need to do this. just build a doppler machine and blow up a welding tank/pipebomb in the red light direction

**Comment by Holinka4ever** [comment]
You know, offtop, but its kinda weird that securities getting new balistic weapon, after opposite group loosing own. Just sayin!

Anyway these is too busted, but in hands of through-wall-seeing antags it will be more dirty as we speak, imagine turned into crit with bleeding in 3 shots from wall. Hilarious!

**Comment by necromanceranne** [comment]
Opposite group? I'm having a bit of trouble following, do you mean like the Syndicate? I'm not sure what removal you're talking about.

**Comment by EnterTheJake** [comment]
> You know, offtop, but its kinda weird that securities getting new balistic weapon, after opposite group loosing own. Just sayin!
> 
> Anyway these is too busted, but in hands of through-wall-seeing antags it will be more dirty as we speak, imagine turned into crit with bleeding in 3 shots from wall. Hilarious!

If this gets merged It might be a good idea to maybe expand tot arsenal to include an automatic, Maybe the C-20r.

**Comment by Loafin34** [comment]
> necromancer anne's response

okay, but all those crew phasic weapons were single shot and had some modicum of effort to make per ammo. The junk round is RNG, the metallic hydrogen bolt requires a crossbow and a competent atmosian who has the time to crystallize hydrogen, and X-ray lasers do like 15 damage. I haven't seen them made AT ALL, because of how crap they are. this is a 2-round burst, bringing it on par with dual laser spam and such. I'd honestly be fine with all of this if the over-engineered wreck gives you a slowdown on par with the pepperspray backpack. That way it retains its superior firepower and gives an incentive to not chase with it.

**Comment by necromanceranne** [comment]
Phasic was in during the test, and I didn't really hear much about it except for a few people concerned about it existing at all, much like right now.

_If_ it somehow proves an issue, and players are being really disruptive chasing phasic, I think that's when it would be time to reconsider it. But evidently, it wasn't proving to be such while it was actually available during the test.

**Comment by necromanceranne** [comment]
Please just say the thing you're talking about rather than being vague. Literally nobody has a consistent culprit of the 'king' of anything was because it changes based on how long you have played. People still pine for tasers.

**Comment by Yobrocharlie** [comment]
This seems like an amazing PR i will thoroughly enjoy using this gun and inevitably finding it as space loot

**Comment by JupiterJaeden** [comment]
I love this, but with all things considered, I feel like 30 damage is a bit high. It really does just make it straight better than a laser- I haven’t experimented with the degradation on this yet, so maybe it’s really bad, but it seems that even with the degradation a 30 round mag of 30 dmg shots is crazy. I’d say 25 damage would be a lot more balanced, and maybe a smaller mag size.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -352,3 +352,198 @@
...
+			fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+		else
+			projectile_speed_multiplier = initial(projectile_speed_multiplier)
+			fire_delay = initial(fire_delay)
+
+	update_appearance()
+
+/// Proc to handle the countdown for our detonation
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/perform_extreme_malfunction(mob/living/user)
+	balloon_alert(user, UNLINT("HOLY FUCK THE GUN IS FUCKING EXPLODING, THROW IT, THROW IT NOW!"))
```
Comment (SmArtKar) [inline code/modules/projectiles/guns/ballistic/automatic.dm:543]:
```suggestion
	balloon_alert(user, UNLINT("HOLY FUCK THE GUN IS GOING TO EXPLODE, THROW IT, THROW IT NOW!"))
```

Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/automatic.dm:543]: yeah, sure

**Comment by necromanceranne** [comment]
Gonna put this on draft for a bit while I work out a cool idea someone gave me.

**Comment by LSommers** [comment]
the sprites look like a midgame gun from the Terraria Calamity Mod and I don't know if I love or hate that

**Comment by necromanceranne** [comment]
That mod has some extremely quality sprites, that's paying Orca a big compliment honestly.

**Comment by necromanceranne** [comment]
Don't quite have enough time to make that change after all

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -52,13 +62,156 @@
...
+	build_type = PROTOLATHE | AWAY_LATHE
+	materials = list(
+		/datum/material/iron = HALF_SHEET_MATERIAL_AMOUNT * 10,
+		/datum/material/bluespace = HALF_SHEET_MATERIAL_AMOUNT,
+	)
+	build_path = /obj/item/ammo_box/magazine/m38/phasic
+	category = list(
+		RND_CATEGORY_WEAPONS + RND_SUBCATEGORY_WEAPONS_AMMO
+	)
+	departmental_flags = DEPARTMENT_BITFLAG_SECURITY
```
Comment (SmArtKar) [inline code/modules/research/designs/weapon_designs.dm:85]: I still stand against phasic rounds being available to the crew, even if they're at the very end of the techweb. Full AP with high damage is bollocks

File: code/modules/research/designs/weapon_designs.dm [hunk 3]
```diff
@@ -52,13 +62,156 @@
...
+	materials = list(
+		/datum/material/iron = HALF_SHEET_MATERIAL_AMOUNT * 30,
+		/datum/material/plastic = HALF_SHEET_MATERIAL_AMOUNT * 3,
+		/datum/material/bluespace = HALF_SHEET_MATERIAL_AMOUNT,
+	)
+	build_path = /obj/item/ammo_box/magazine/m38/phasic
+	category = list(
+		RND_CATEGORY_WEAPONS + RND_SUBCATEGORY_WEAPONS_AMMO
+	)
+	departmental_flags = DEPARTMENT_BITFLAG_SECURITY
```
Comment (SmArtKar) [inline code/modules/research/designs/weapon_designs.dm:197]: Same here

**Review by SmArtKar — CHANGES_REQUESTED** [review]

File: code/modules/research/designs/weapon_designs.dm (hunk 2 above)
Comment (SmArtKar) [reply code/modules/research/designs/weapon_designs.dm:85]: For the record, 5 shots to crit anybody is less than 3 bursts (20 damage per shot) which is pretty strong compared to lasers which have 25 damage with 0 AP ***and*** need recharging (which makes them inherently weaker than mag-based weapons)

Comment (necromanceranne) [reply code/modules/research/designs/weapon_designs.dm:85]: Maybe locked behind illegal tech, like someone suggested?

Comment (necromanceranne) [reply code/modules/research/designs/weapon_designs.dm]: resolved this by removing phasic and making true strike bounce a lot, and accurately, but reducing the damage by a lot (15).

**Comment by necromanceranne** [comment]
image

I put the suppressor compatibility back in.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -352,3 +352,192 @@
...
+	else
+		. += span_notice("\The [src] is in the process of system degradation. It is currently at stage [degradation_stage] of [degradation_stage_max]. Use a multitool on [src] to recalibrate. Alternatively, insert it into a weapon recharger.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/update_icon_state()
+	. = ..()
+	if(!shots_before_degradation)
+		inhand_icon_state = "[base_icon_state]-empty"
+	else
+		inhand_icon_state = "[base_icon_state]"
+	return ..()
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:439]: Calls parent twice here

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 5]
```diff
@@ -352,3 +352,192 @@
...
+		It is possible for the weapon to suffer from unintended discombulations due to closed heat distribution systems should the weapon be tampered with. \
+		R&D are working on this issue before the weapon sees commercial sales. That, and trying to work out why the weapon's onboard computation systems suffer \
+		from so many calculation errors.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/examine(mob/user)
+	. = ..()
+	if(shots_before_degradation)
+		. += span_notice("\The [src] can fire [shots_before_degradation] more times before risking system degradation.")
+	else
+		. += span_notice("\The [src] is in the process of system degradation. It is currently at stage [degradation_stage] of [degradation_stage_max]. Use a multitool on [src] to recalibrate. Alternatively, insert it into a weapon recharger.")
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:431]:
```suggestion
	if(shots_before_degradation)
		. += span_notice("[src] can fire [shots_before_degradation] more times before risking system degradation.")
	else
		. += span_notice("[src] is in the process of system degradation. It is currently at stage [degradation_stage] of [degradation_stage_max]. Use a multitool on [src] to recalibrate. Alternatively, insert it into a weapon recharger.")
```
Puts in `The` for you by default

Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:431]: Also, screentips?

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 6]
```diff
@@ -352,3 +352,192 @@
...
+	return TRUE
+
+/obj/item/gun/ballistic/automatic/battle_rifle/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!tool.use_tool(src, user, 20 SECONDS, volume = 50))
+		balloon_alert(user, "interrupted!")
+		return
+
+	if(emp_malfunction)
+		emp_malfunction = FALSE
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:472]:
```suggestion
	emp_malfunction = FALSE
```
Gonna end up false at the end of the day anyways

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 7]
```diff
@@ -352,3 +352,192 @@
...
+
+	shots_before_degradation = initial(shots_before_degradation)
+	degradation_stage = initial(degradation_stage)
+	projectile_speed_multiplier = initial(projectile_speed_multiplier)
+	fire_delay = initial(fire_delay)
+	update_appearance()
+	balloon_alert(user, "system reset.")
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:479]:
```suggestion
	balloon_alert(user, "system reset")
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 8]
```diff
@@ -352,3 +352,192 @@
...
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return FALSE
+	obj_flags |= EMAGGED
+	projectile_damage_multiplier = emagged_projectile_damage_multiplier
+	balloon_alert(user, "heat distribution systems deactivated")
+	return TRUE
+
+/obj/item/gun/ballistic/automatic/battle_rifle/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:466]: Parent call does nothing

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 9]
```diff
@@ -352,3 +352,192 @@
...
+	if(!tool.use_tool(src, user, 20 SECONDS, volume = 50))
+		balloon_alert(user, "interrupted!")
+		return
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:469]:
```suggestion
		return ITEM_INTERACT_BLOCKING
```

File: code/modules/projectiles/guns/ballistic/automatic.dm (hunk 7 above)
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:479]:
```suggestion
	balloon_alert(user, "system reset.")
	return ITEM_INTERACT_SUCCESS
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 10]
```diff
@@ -352,3 +352,192 @@
...
+	shots_before_degradation = initial(shots_before_degradation)
+	degradation_stage = initial(degradation_stage)
+	projectile_speed_multiplier = initial(projectile_speed_multiplier)
+	fire_delay = initial(fire_delay)
+	update_appearance()
+	balloon_alert(user, "system reset.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/try_fire_gun(atom/target, mob/living/user, params)
+	. = ..()
+	if(!chambered || chambered && !chambered.loaded_projectile)
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:483]: Chaining operators here in a scary way, be sure to use parenthesis
```suggestion
	if(!chambered || (chambered && !chambered.loaded_projectile))
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 11]
```diff
@@ -352,3 +352,192 @@
...
+		return
+
+	if(shots_before_degradation)
+		shots_before_degradation --
+		return
+
+	else if (obj_flags & EMAGGED && degradation_stage == degradation_stage_max && !explosion_timer)
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:490]:
```suggestion
	else if ((obj_flags & EMAGGED) && degradation_stage == degradation_stage_max && !explosion_timer)
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 12]
```diff
@@ -352,3 +352,192 @@
...
+		perform_extreme_malfunction(user)
+
+	else
+		attempt_degradation(FALSE)
+
+
+/obj/item/gun/ballistic/automatic/battle_rifle/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(chambered.loaded_projectile && emp_malfunction && prob(75) || chambered.loaded_projectile && degradation_stage == degradation_stage_max && prob(75))
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:498]: Same here with the parenthesis
```suggestion
	if((chambered.loaded_projectile && emp_malfunction && prob(75)) || (chambered.loaded_projectile && degradation_stage == degradation_stage_max && prob(75)))
```
But also it's a bit more complex than it needs to be
```suggestion
	if(chambered.loaded_projectile && prob(75) && (emp_malfunction || degradation_stage == degradation_stage_max))
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 13]
```diff
@@ -352,3 +352,192 @@
...
+	degradation_stage = clamp(degradation_stage + (obj_flags & EMAGGED ? 2 : 1), 0, degradation_stage_max)
+	projectile_speed_multiplier = clamp(initial(projectile_speed_multiplier) + degradation_stage * 0.1, initial(projectile_speed_multiplier), maximum_speed_malus)
+	fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+	do_sparks(1, TRUE, src)
+	update_appearance()
+
+/// Called by /obj/machinery/recharger while inserted: attempts to recalibrate our gun but reducing degradation.
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/attempt_recalibration(restoring_shots_before_degradation = FALSE, recharge_rate = 1)
+	if(emp_malfunction)
+		emp_malfunction = FALSE
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:519]:
```suggestion
	emp_malfunction = FALSE
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 14]
```diff
@@ -352,3 +352,192 @@
...
+			fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+		else
+			projectile_speed_multiplier = initial(projectile_speed_multiplier)
+			fire_delay = initial(fire_delay)
+
+	update_appearance()
+
+/// Proc to handle the countdown for our detonation
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/perform_extreme_malfunction(mob/living/user)
+	balloon_alert(user, UNLINT("HOLY FUCK THE GUN IS GOING TO EXPLODE, THROW IT, THROW IT NOW!"))
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:537]: Little goofy

If you had a problem with people not noticing it was going to explode, I would maybe recommend looping through alerts before it blows

```dm
balloon_alert("gun is exploding!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding, throw it!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding, throw it!") 
// (1 second timer)
if(!thrown)
	balloon_alert("GUN IS ABOUT TO EXPLODE!") 
// 1 second timer
explode()
```

Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/automatic.dm:537]: It was mostly just as a joke more than any practical purpose.

Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/automatic.dm]: I've at the very least shortened the message.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 15]
```diff
@@ -352,3 +352,196 @@
...
+			fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+		else
+			projectile_speed_multiplier = initial(projectile_speed_multiplier)
+			fire_delay = initial(fire_delay)
+
+	update_appearance()
+
+/// Proc to handle the countdown for our detonation
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/perform_extreme_malfunction(mob/living/user)
+	balloon_alert(user, UNLINT("[src] is exploding, throw it!"))
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:541]:
```suggestion
	balloon_alert(user, "gun is exploding, throw it!")
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 16]
```diff
@@ -352,3 +352,196 @@
...
+
+/obj/item/gun/ballistic/automatic/battle_rifle/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/scope, range_modifier = 2)
+	register_context()
+
+/obj/item/gun/ballistic/automatic/battle_rifle/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	if(held_item.tool_behaviour == TOOL_MULTITOOL && shots_before_degradation < max_shots_before_degradation)
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:416]:
```suggestion
	if(held_item?.tool_behaviour == TOOL_MULTITOOL && shots_before_degradation < max_shots_before_degradation)
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 17]
```diff
@@ -352,3 +352,196 @@
...
+		context[SCREENTIP_CONTEXT_LMB] = "Reset System"
+	return CONTEXTUAL_SCREENTIP_SET
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/automatic.dm:418]:
```suggestion
		return CONTEXTUAL_SCREENTIP_SET
```

**Review by MrMelbert — APPROVED** [review]

**Comment by LSommers** [comment]
I've got to say, I'm seeing this almost exclusively used by command powergamers rather than sec during literally any situation. It seems considerably more powerful than any other station firearm, which is possibly _why_.

**Pull request merged**